local-cli/runAndroid/runAndroid.js

```diff
@@ -156,6 +156,20 @@ function buildAndRun(args) {
       'utf8'
     ).match(/package="(.+?)"/)[1];
 
+    const variantName = args.variant || 'debug';
+    const gradleFile = fs.readFileSync('app/build.gradle', 'utf8');
+
+    const appId = gradleFile.match(/applicationId "(.+?)"/)[1];
+    const buildTypes = gradleFile.substr(gradleFile.indexOf('buildTypes'));
+    const buildLoc = buildTypes.indexOf(variantName);
+    let appIdSuffix = '';
+    if (buildLoc >= 0) {
+        const buildSectionPlus = buildTypes.substr(buildLoc);
+        const buildSection = buildSectionPlus.substr(0, buildSectionPlus.indexOf('}'));
+        appIdSuffix = buildSection.match(/applicationIdSuffix "(.+?)"/)[1];
+    }
+
```

Comment:
Hi.  Thanks for your feedback.  Let me just make sure I understand what you're proposing
- appIdSuffix, an optional arg to react-native run-android so developers can specify this value manually
- getAppIdFromGradle, move the above code to a separate file for reuse in other areas.  This also becomes the default if no value is specified on the command line.  Note that I like that last part because it means less thinking every time you build your app.  Variant implies suffix (assuming our simple gradle parsing succeeds).